src/BoundaryConditions/continuous_boundary_function.jl

```diff
@@ -124,7 +124,7 @@ const ZBoundaryFunction{LX, LY, S} = ContinuousBoundaryFunction{LX, LY, Nothing,
 
 # Return ContinuousBoundaryFunction on east or west boundaries.
 @inline function getbc(cbf::XBoundaryFunction{LY, LZ, S}, j::Integer, k::Integer,
-                       grid::AbstractGrid, clock, model_fields, args...) where {LY, LZ, S}
+                       grid::AbstractGrid, clock, model_fields) where {LY, LZ, S}
```

Comment:
I suggest keeping the original behavior so that future model developers can support more arguments?

Reply:
If we want this, we need to pass the args... to the continuous boundary function, which is not the original behavior. If we want to support more arguments, there is some structural work to do on the `ContinuousBoundaryFunctions` to make sure that field dependencies do not conflict with the extra arguments passed to the `getbc`

Reply:
Here, we want to have a generic `args...` so that a model implementation can pass additional arguments into `fill_halo_regions!` (which then are propagated into `getbc`). However, having `args...` does _not_ imply that `ContinuousBoundaryFunction` has to support interpreting it. `ContinuousBoundaryFunction` can have an interface which only utilizes the first two arguments `clock, model_fields` and discards the rest. Then the extra `args...` are only used by `DiscreteBoundaryFunction`.